cargo-near/src/commands/build_command/docker.rs

```diff
@@ -239,4 +242,180 @@ impl super::BuildCommand {
+            None
+        } else {
+            Some(contract_path)
+        };
+
+        let source_code_snapshot = source_id::SourceId::for_git(
+            &docker_build_meta.source_code_git_url,
+            source_id::GitReference::Rev(cloned_repo.initial_crate_in_repo.head.to_string()),
+        )
+        .map_err(|err| color_eyre::eyre::eyre!("compute SourceId {}", err))?;
+        Ok(Self {
+            build_environment,
+            contract_path,
+            source_code_snapshot,
+        })
+    }
+
+    fn docker_args(&self) -> Vec<String> {
+        let mut result = vec![
+            "--env".to_string(),
+            format!("{}={}", INSIDE_DOCKER_ENV_KEY, self.build_environment),
+            "--env".to_string(),
+            format!(
+                "{}={}",
+                SOURCE_CODE_SNAPSHOT_ENV_KEY,
+                self.source_code_snapshot.as_url()
+            ),
+        ];
+
+        if let Some(ref contract_path) = self.contract_path {
+            result.extend(vec![
+                "--env".to_string(),
+                format!("{}={}", CONTRACT_PATH_ENV_KEY, contract_path),
+            ]);
+        }
+
+        result
+    }
+}
+struct EnvVars {
+    build_info: Nep330BuildInfo,
+    rust_log: String,
+    repo_link: Option<String>,
+    revision: String,
+}
+
+impl EnvVars {
+    fn new(
+        docker_build_meta: &metadata::ReproducibleBuild,
+        cloned_repo: &cloned_repo::ClonedRepo,
+    ) -> color_eyre::eyre::Result<Self> {
+        let build_info = Nep330BuildInfo::new(docker_build_meta, cloned_repo)?;
+        let repo_link = cloned_repo.crate_metadata().root_package.repository.clone();
+        let revision = cloned_repo.initial_crate_in_repo.head.to_string();
+        Ok(Self {
+            build_info,
+            rust_log: RUST_LOG_EXPORT.to_string(),
+            repo_link,
+            revision,
+        })
+    }
+
+    fn docker_args(&self) -> Vec<String> {
+        let mut result = self.build_info.docker_args();
+
+        if let Some(repo_link_hint) = self.compute_repo_link_hint() {
+            result.extend(vec![
+                "--env".to_string(),
+                format!("{}={}", REPO_LINK_HINT_ENV_KEY, repo_link_hint,),
+            ]);
+        }
+        result.extend(vec!["--env".to_string(), self.rust_log.clone()]);
+        result
+    }
+    fn compute_repo_link_hint(&self) -> Option<String> {
+        let url = self
+            .repo_link
+            .clone()
+            .and_then(|repo_link| url::Url::parse(&repo_link).ok());
+
+        let url = url
+            .clone()
+            .and_then(|url| url.host_str().map(ToString::to_string))
+            .filter(|host| *host == "github.com")
+            .and(url);
+        let commit_hint = url.and_then(|url| {
+            let existing_path = url.path();
+            url.join(&format!("{}/tree/{}", existing_path, self.revision))
+                .ok()
+        });
+        commit_hint.map(|url| url.to_string())
```

Comment:
I believe this code is easier to read:

```suggestion
        let url = url::Url::parse(&self.repo_link?).ok()?;

        if url.host_str() == "github.com" {
            let existing_path = url.path();
            Some(url.join(&format!("{}/tree/{}", existing_path, self.revision))
                .ok()?
                .to_string())
        } else {
            None
        }
```

Reply:
fixed by https://github.com/near/cargo-near/commit/db93a3c1e8fe2cdefcd77f46efc886e17b7b0ad4